app/bundles/ReportBundle/Controller/ReportController.php

```diff
@@ -630,8 +630,20 @@ public function viewAction($objectId, $reportPage = 1)
 
         // Setup dynamic filters
         $filterDefinitions = $model->getFilterList($entity->getSource());
-        $dynamicFilters    = $session->get('mautic.report.'.$objectId.'.filters', []);
-        $filterSettings    = [];
+        /** @var array $dynamicFilters */
+        $dynamicFilters = $session->get('mautic.report.'.$objectId.'.filters', []);
+        $filterSettings = [];
+
+        if (count($dynamicFilters) > 0 && count($entity->getFilters()) > 0) {
+            foreach ($entity->getFilters() as $fid => $filter) {
+                foreach ($dynamicFilters as $dfcol => &$dfval) {
+                    if (1 === $filter['dynamic'] && $filter['column'] === $dfcol) {
+                        $dfval['expr'] = $filter['condition'];
```

Comment:
1) Are you sure you want to use same condition from filter?
2) Maybe add "break" to if condition?